Core/src/com/serotonin/m2m2/db/H2Proxy.java

```diff
@@ -127,127 +108,82 @@ private Path getDbPathFromUrl(String propertyPrefix) {
+            //Create a reference for the dump file
+            String dumpFileName = "mah2.h2.196.sql.zip";
+            Path dumpPath = Common.MA_HOME_PATH.resolve("backup").resolve(dumpFileName);
             try {
-                //Did a previous attempt fail?
-                if(bakDbPath.toFile().exists()){
-                    //TODO is this logic ideal?
-                    //Failed attempt assume the backup is the best one to keep so replace the existing
-                    Files.copy(bakDbPath, existingDbPath);
-                }else {
-                    //Move existing db (for safety)
-                    Files.copy(existingDbPath, bakDbPath);
-                }
+                LOG.info("Dumping legacy database to file " + dumpPath.toString());
+                dump(legacy, dumpPath);
+                
+                //Delete existing so we can re-create it using the dump script
+                Files.delete(legacy);
 
-                //First we will try to get the version from the existing db to make sure 
-                org.h2.Driver.load();
-                String fullUrl = getUrl(propertyPrefix);
                 String user = Common.envProps.getString(propertyPrefix + "db.username", null);
                 String password = Common.envProps.getString(propertyPrefix + "db.password", null);
-
-                try(Connection conn = DriverManager.getConnection(fullUrl, user, password);){
-                    Statement stat = conn.createStatement();
-                    ResultSet rs = stat.executeQuery(H2_CREATE_VERSION_SELECT);
-                    if(rs.next()) {
-                        int version = rs.getInt(1);
-                        if(version > h2Version) {
-                            // This would be when we don't need to do the upgrade but no properties file 
-                            // was found? i.e. someone deleted it or edited it poorly
-                            h2Properties.put("h2Version", Integer.toString(version));
-                            deleteTempFiles = true;
-                            saveProperties = true;
-                            LOG.info("H2 database is version " + version + " , not upgrading.");
-                            return;
-                        }
-                        LOG.info("H2 database is version " + version + " , will upgrade to " + h2Version + ".");
-                    }
-                    LOG.info("Exporting exisiting H2 database...");
-                    stat.executeQuery("SCRIPT DROP TO '" + dumpPath.toString() + "' COMPRESSION ZIP");
-                    stat.close();
-                }
                 
-                //Delete existing so we can re-create it using the dump script
-                Files.delete(existingDbPath);
-
+                String url = getUrl(propertyPrefix);
                 //Open a connection and import the dump script
                 LOG.info("Importing exisiting H2 database...");
```

Comment:
this "exisiting" typo is in two retained log messages

Reply:
Fixed